crates/matrix-sdk-crypto/src/olm/group_sessions/inbound.rs

```diff
@@ -359,7 +375,10 @@ impl InboundGroupSession {
     /// Has the session been imported from a file or server-side backup? As
     /// opposed to being directly received as an `m.room_key` event.
     pub fn has_been_imported(&self) -> bool {
-        self.imported
+        match self.key_source {
+            KeySource::Direct => false,
+            _ => true,
```

Comment:
Is this the right interpretation?  Should we also add a function that exposes `key_source`?